plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/LibvirtComputingResource.java

```diff
@@ -2337,8 +2337,9 @@ public int compare(final DiskTO arg0, final DiskTO arg1) {
                     disk.setCacheMode(DiskDef.DiskCacheMode.valueOf(volumeObjectTO.getCacheMode().toString().toUpperCase()));
                 }
             }
-
-            vm.getDevices().addDevice(disk);
+            if (vm.getDevices() != null) {
```

Comment:
Should not an exception be thrown in this case?

Reply:
I only mentioned an exception for a single reason. Right now the code works; and, if we ever get a null calling `vm.getDevices() `, a run time exception will be thrown, and we will be able to easily pin point the problem. However, after applying your code, for sure something else will break or not work as expected, which will make more difficult for us to pin point the problem later (if it ever happens).

Reply:
hum. I mean another three "vm.getDevices;" are all in the same function,  vm.getDevices can always return same object, so add checker at  line 2340 mean that check the parameter vm is legal.

Reply:
Actually, I am not talking about the variable `vm`. I am instead, referring to the return of `vm.getDevices()` method.

Reply:
i got, the exception stack can help us where error happens accurately .